packages/compass-aggregations/src/modules/count-documents.ts

```diff
@@ -0,0 +1,106 @@
+import type { Reducer } from 'redux';
+import type { AggregateOptions } from 'mongodb';
+import type { ThunkAction } from 'redux-thunk';
+import type { RootState } from '.';
+import { DEFAULT_MAX_TIME_MS } from '../constants';
+import { generateStage } from './stage';
+import type { Actions as WorkspaceActions } from './workspace';
+import { ActionTypes as WorkspaceActionTypes } from './workspace';
+
+
+export enum ActionTypes {
+  CountStarted = 'compass-aggregations/countStarted',
+  CountFinished = 'compass-aggregations/countFinished',
+  CountFailed = 'compass-aggregations/countFailed',
+};
+
+type CountStartedAction = {
+  type: ActionTypes.CountStarted;
+};
+
+type CountFinishedAction = {
+  type: ActionTypes.CountFinished;
+  count: number;
+};
+
+type CountFailedAction = {
+  type: ActionTypes.CountFailed;
+};
+
+export type Actions =
+  | CountStartedAction
+  | CountFinishedAction
+  | CountFailedAction;
+
+export type State = {
+  count?: number;
+  loading: boolean;
+};
+
+export const INITIAL_STATE: State = {
+  loading: false,
+};
+
+const reducer: Reducer<State, Actions | WorkspaceActions> = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case WorkspaceActionTypes.WorkspaceChanged:
+      return INITIAL_STATE;
```

Comment:
As we are not cleaning up inflight counts when workspace type is changed (both for count and for results), you can run into a pretty nasty race condition where slower, older pipeline results and counts replace content of the newer one. Here's how it looks in UI:

![Kapture 2022-04-21 at 18 16 46](https://user-images.githubusercontent.com/5036933/164505872-b2c02b8f-99dd-4a49-8569-ebe6ea3770b7.gif)

We should definitely fix that both for counts and for results